### ISSUE TITLE
Support relative path with T4 template path

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/EFCorePowerTools#1876]

**Comment by ErikEJ** [comment]
@ClaudioBiagioli FYI

**Comment by ErikEJ** [comment]
I implemented a fix for this in the [latest daily build](https://www.vsixgallery.com/extension/f4c4712c-ceae-4803-8e52-0e2049d5de9f), would be grateful if you could try it out.

**Comment by ClaudioBiagioli** [comment]
Hi Erik.
I tried with 2.5.1548 but with
"T4TemplatePath": "..\\.."
the CodeTemplates folder is still created in the root folder.

C.

**Comment by ErikEJ** [comment]
@ClaudioBiagioli  OK, maybe there is a bug in the drop location resolution, but are the templates used from the correct folder?

**Comment by ErikEJ** [comment]
@ClaudioBiagioli I actually think it should be your responsibility to place the templates, not up to the tool if you have a custom template path. I will change that.

**Comment by ClaudioBiagioli** [comment]
Yes Erik, but, since the tool creates the templates when they're not present, I suppose that it takes the wrong location for some reason.
I have a CodeTemplates in the supposed location but it was not considered and the new folder was created in the root with default values.

**Comment by ErikEJ** [comment]
@ClaudioBiagioli I have just changed it so the tool no longer creates anything if T4 template folder is specified. I will try to repro.

**Comment by ErikEJ** [comment]
@ClaudioBiagioli OK, getting a relative path is not trivial, I found a bug.

**Comment by ErikEJ** [comment]
@ClaudioBiagioli This now works for me:

```json
"T4TemplatePath": "..\\..\\"
```

**Comment by ClaudioBiagioli** [comment]
Yes! It's working as expected now! 👍 
For a complete User experience I would add the possibility to set the folder location also in the Advanced options form, may be after the DBContext Path/namespace fields.

**Comment by ErikEJ** [comment]
Yeah, I will add that when 2 other users ask for it...

**Comment by ClaudioBiagioli** [comment]
Hello Erik, at last I'm back with this issue, since I had some spare time to check the code and I'm ready to submit a small PR for add the T4 template path into the advanced dialog box, if you think it could be useful (I think it is).

Please let me know.

**Comment by ErikEJ** [comment]
@ClaudioBiagioli Sure, let's do it. The feature is completely invisible now anyway.
